library/src/main/java/org/apache/apex/malhar/lib/state/managed/AbstractManagedStateImpl.java

```diff
@@ -203,7 +203,12 @@ public void setup(OperatorContext context)
       //delete all the wal files with windows > activationWindow.
       //All the wal files with windows <= activationWindow are loaded and kept separately as recovered data.
       try {
-        for (long recoveredWindow : checkpointManager.getWindowIds(operatorContext.getId())) {
+        long[] recoveredWindows = checkpointManager.getWindowIds(operatorContext.getId());
+        if (recoveredWindows == null) {
+          readerService = Executors.newFixedThreadPool(numReaders, new NameableThreadFactory("managedStateReaders"));
```

Comment:
This is already being done at the end. why do you need to duplicate the stmt here?